### DISCUSSION TITLE
Is it possible to have different step in NumberField and NumberFieldDecrement/Increment?

### DISCUSSION THREAD

**Discussion by vas11yev1work** [unovue/reka-ui#1194] [Q&A]
Hello! I use shadcn-vue, but I notice this problem in radix-vue.
I want to be able to enter a deciman number (e.g. 0.05) in the input field, but at the same time, when pressing the inc/dec buttons, increase/decrease the value not by 0.01

```vue
<NumberField
  locale="ru-RU"
  :min="1"
  :default-value="undefined"
  :format-options="{ style: 'currency', currency: 'RUB' }"
  :step="0.01" // if i set 1, i will not be able to set a decimal value into the input
  @update:model-value="setNumber('priceWithDiscount', $event)"
>
  <NumberFieldContent>
    <NumberFieldDecrement /> // if i click here, value decrease by 0.01
    <FormControl>
      <NumberFieldInput />
    </FormControl>
    <NumberFieldIncrement /> // if i click here, value increase by 0.01
  </NumberFieldContent>
</NumberField>
```

**Comment by sadeghbarati** [comment]
PR https://github.com/radix-vue/radix-vue/pull/1202 (I think I'm not doing it right let's see other maintainers' reviews)

After this PR gets merged you do something like
I see you are using `shadcn-vue` you need to pass/forward `radix-vue` **slotProps** into your `NumberField` component `slot` too


```html
<NumberField
  locale="ru-RU"
  :min="1"
  :default-value="undefined"
  :format-options="{ style: 'currency', currency: 'RUB' }"
  :step="0.01" // if i set 1, i will not be able to set a decimal value into the input
  @update:model-value="setNumber('priceWithDiscount', $event)"
  v-slot="{ handleIncrease, handleDecrease }"
>
  <NumberFieldContent>
    <NumberFieldDecrement /> // decrease by 0.01
    <FormControl>
      <NumberFieldInput />
    </FormControl>
    <NumberFieldIncrement /> // increase by 0.01
  </NumberFieldContent>
  
  
  <Button @click="handleIncrease(0.05)">Custom Button increase by 0.05</Button>
</NumberField>
```

**Reply by vas11yev1work** [reply]
Why is your PR closed?

**Reply by sadeghbarati** [reply]
Sry I didn't mention why I closed it, that PR was not the right fix, and also the press and hold to +/- is not working with this approach

**Comment by zernonia** [comment]
Unfortunately currently `NumberField` doesn't support `step` and  `formatOptions="{   minimumFractionDigits: 2,   maximumFractionDigits: 3 }"` for now. 

However I also noticed a small bug were you were unable to set 3 fraction digit, it always clamp the value up.
